### PR TITLE
LUN-81: Check for alternative article

### PR DESCRIPTION
**Pull request by SteffiStuffel**
This pull request belongs to an issue on our [bugtracker](https://issues.tuerantuer.org/secure/RapidBoard.jspa?rapidView=27&view=detail).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword _LUN_.

### CODE REVIEW

**Review by klinzo — COMMENTED** [review]
Do you have an example where it is the case that only the article differs?

**Comment by SteffiStuffel** [comment]
> Do you have an example where it is the case that only the article differs?

Metall, Elektro, Maschinen -> Grundlagen Elektronik -> Der Spachtel / Die Spachtel

**Comment by klinzo** [comment]
Hmm okay... wouldn't it make more sense to just highlight both options as valid then? I.e. if an option is selected, both are highlighted green? At least I would be confused otherwise why one time `die` and the other time `der` is shown as correct. So correctAnswer would become a correctAnswers array.

**Comment by SteffiStuffel** [comment]
> Hmm okay... wouldn't it make more sense to just highlight both options as valid then? I.e. if an option is selected, both are highlighted green? At least I would be confused otherwise why one time `die` and the other time `der` is shown as correct. So correctAnswer would become a correctAnswers array.

Lets wait for a design / input here, how this should be handled, an keep it like this for now.

File: src/screens/choice-exercises/components/SingleChoiceExercise.tsx [hunk 1]
```diff
@@ -29,31 +29,44 @@ interface SingleChoiceExercisePropsType {
...
+  const isCorrectAlternative = (selectedAnswer: Answer): boolean => {
+    let found = false
+    currentDocument.alternatives.forEach(alternative => {
+      if (alternative.article === selectedAnswer.article && alternative.alt_word === selectedAnswer.word) {
+        setCorrectAnswer(selectedAnswer)
+        found = true
+      }
+    })
+    return found
+  }
```
Comment (klinzo) [inline src/screens/choice-exercises/components/SingleChoiceExercise.tsx:64]:
```suggestion
  const correctAlternatives: Answer[] = currentDocument.alternatives.map(it => ({
    article: it.article,
    word: it.alt_word
  }))
```

File: src/screens/choice-exercises/components/SingleChoiceExercise.tsx [hunk 2]
```diff
@@ -29,31 +29,44 @@ interface SingleChoiceExercisePropsType {
...
+
   const onClickAnswer = (selectedAnswer: Answer) => {
     setSelectedAnswer(selectedAnswer)
 
-    if (isAnswerEqual(selectedAnswer, correctAnswer)) {
+    if (isAnswerEqual(selectedAnswer, correctAnswer) || isCorrectAlternative(selectedAnswer)) {
```
Comment (klinzo) [inline src/screens/choice-exercises/components/SingleChoiceExercise.tsx:69]:
```suggestion
    const correctSelected = [correctAnswer, ...correctAlternatives].find(it => isAnswerEqual(it, selectedAnswer)) ?? null
    if (correctSelected !== null) {
      setCorrectAnswer(correctSelected)
```

Comment (SteffiStuffel) [reply src/screens/choice-exercises/components/SingleChoiceExercise.tsx]: done

File: src/screens/choice-exercises/components/SingleChoiceExercise.tsx (hunk 1 above)
Comment (SteffiStuffel) [reply src/screens/choice-exercises/components/SingleChoiceExercise.tsx:64]: done

**Pull request merged**